linera-core/src/worker.rs

```diff
@@ -1130,6 +1227,8 @@ where
         // Verify that all required bytecode hashed certificate values are available, and no unrelated ones provided.
         self.check_no_missing_bytecode(block, hashed_certificate_values)
             .await?;
+
+        self.check_no_missing_blob(block, hashed_blobs).await?;
```

Comment:
If we don't save the blobs here, this makes multi-owner chains vulnerable to an attack by a malicious owner:

* Propose a valid block with a blob.
* The validators all vote to validate it.
* Submit the `ValidatedBlock` certificate.
* The validators all vote to confirm it.
* Delete the blob.
* Now the chain cannot make progress anymore because nobody has the blob but they all voted to confirm the block that requires it.

I think it's sufficient to put these into the chain manager's pending blobs when voting to validate or confirm the block (probably in `create_vote`):

Then if a quorum votes to validate the block that means more than a third of the validators are both honest and have the blob, until a block at this height is confirmed.